optimum/habana/transformers/generation/utils.py

```diff
@@ -735,7 +735,7 @@ def generate(
         # if generation_config.bucket_size <= 0, padding is handled by the generating fn (like greedy_search)
         if generation_config.static_shapes and generation_config.bucket_size > 0:
             assert (
-                generation_mode == GenerationMode.GREEDY_SEARCH or generation_mode == GenerationMode.BEAM_SEARCH
+                generation_mode == GenerationMode.GREEDY_SEARCH or generation_mode == GenerationMode.SAMPLE
```

Comment:
Lets have the check of BEAM SEARCH too since bucketing changes from Sayantan works in beam search.